File: 6-kyu/persistent-bugger/javascript/solution.js
let iteration = -1
function multiply(multiplier, before) {
  return multiplier * before
}
function multiplicatePersistence(num) {
  if (num.length == 1)
  {
    return num;
  }
  iteration++;
  return multiplicatePersistence([...String(num)].reduce(multiply))
}
function persistence(num) {
   iteration = -1
   multiplicatePersistence(num)
   return iteration
}

File: 6-kyu/persistent-bugger/ruby/solution.rb
$iterator = -1
def multiply(num)
  if (num.to_s.length == 1)
    return num
  end
  $iterator+=1
  self.multiply(num.to_s.split("").map(&:to_i).inject(:*))
end
def persistence(n)
    $iterator = 0
    self.multiply(n)
    return $iterator
end